File: android/patient_reception/app/src/main/java/jp/ac/tottori_u/med/hosp/patient_reception/models/object/CheckObject.java
package jp.ac.tottori_u.med.hosp.patient_reception.models.object;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by nishimura on 2018/10/03.
 */

public class CheckObject extends AbstractObject {

    public static String CHECK_FLAG = "check_flag";
    public static String BUTTON = "button";

    private String uniqueId;
    private boolean checkFlg;
    private String button;

    public String getButton() {
        return button;
    }

    public void setButton(String button) {
        this.button = button;
    }

    public String getUniqueId() {
        return uniqueId;
    }

    public void setUniqueId(String uniqueId) {
        this.uniqueId = uniqueId;
    }

    public Boolean isCheckFlg() {
        return checkFlg;
    }

    public void setCheckFlg(Boolean checkFlg) {
        this.checkFlg = checkFlg;
    }

    @Override
    protected void setParameters(JSONObject json) throws JSONException {
        // CheckFlg
        this.checkFlg = json.isNull(CHECK_FLAG)?false:json.getBoolean(CHECK_FLAG);

        this.button = json.isNull(BUTTON)?"登録":json.getString(BUTTON);
    }
}

File: ios/patient_reception/patient_reception/Models/ModelObject/CheckObject.swift
//
//  CheckObject.swift
//  patient_reception
//
//  Created by nishimura on 2018/09/27.
//  Copyright © 2018年 jp.ac.tottori-u.med.hosp. All rights reserved.
//

import UIKit
import SwiftyJSON

class CheckObject: AbstractObject {
    var uniqueId:String?
    var checkFlg:Bool = false
    var button:String?
    
    override func abstractSetData(json: JSON) {
        
        let r = json[Response.checkFlag.rawValue].bool
        if let r = r {
            self.checkFlg = r
        }
        self.button = json[Response.button.rawValue].string
    }
}
